influxql/select.go

```diff
@@ -199,11 +204,266 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions) (Iter
+		}
+		input, ok := rhs.(FloatIterator)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected rhs to be FloatIterator, got %T", rhs)
+		}
+		return &floatTransformIterator{
+			input: input,
+			fn: func(p *FloatPoint) *FloatPoint {
+				if p == nil {
+					return nil
+				}
+				p.Value = fn(lit.Val, p.Value)
+				return p
+			},
+		}, nil
+	case func(float64, float64) bool:
+		lit, ok := lhs.(*NumberLiteral)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected lhs to be NumberLiteral, got %T", lhs)
+		}
+		input, ok := rhs.(FloatIterator)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected lhs to be FloatIterator, got %T", rhs)
+		}
+		return &floatBoolTransformIterator{
+			input: input,
+			fn: func(p *FloatPoint) *BooleanPoint {
+				if p == nil {
+					return nil
+				}
+				return &BooleanPoint{
+					Name:  p.Name,
+					Tags:  p.Tags,
+					Time:  p.Time,
+					Value: fn(lit.Val, p.Value),
+					Aux:   p.Aux,
+				}
+			},
+		}, nil
+	}
+	return nil, fmt.Errorf("unable to construct transform iterator from %T and %T", lhs, rhs)
+}
+
+func buildTransformIterator(lhs Iterator, rhs Iterator, op Token, ic IteratorCreator, opt IteratorOptions) (Iterator, error) {
+	fn := getBinaryExprFunc(getIteratorDataType(lhs), op)
+	switch fn := fn.(type) {
+	case func(float64, float64) float64:
+		left, ok := lhs.(FloatIterator)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected lhs to be FloatIterator, got %T", lhs)
+		}
+		right, ok := rhs.(FloatIterator)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected lhs to be FloatIterator, got %T", rhs)
+		}
+		return &floatTransformIterator{
+			input: left,
+			fn: func(p *FloatPoint) *FloatPoint {
+				if p == nil {
+					return nil
+				}
+				p2 := right.Next()
+				if p2 == nil {
+					return nil
+				}
+				p.Value = fn(p.Value, p2.Value)
+				return p
+			},
+		}, nil
+	case func(float64, float64) bool:
+		left, ok := lhs.(FloatIterator)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected lhs to be FloatIterator, got %T", lhs)
+		}
+		right, ok := rhs.(FloatIterator)
+		if !ok {
+			return nil, fmt.Errorf("type mismatch, expected lhs to be FloatIterator, got %T", rhs)
+		}
+		return &floatBoolTransformIterator{
+			input: left,
+			fn: func(p *FloatPoint) *BooleanPoint {
+				if p == nil {
+					return nil
+				}
+				p2 := right.Next()
+				if p2 == nil {
+					return nil
+				}
+				return &BooleanPoint{
+					Name:  p.Name,
+					Tags:  p.Tags,
+					Time:  p.Time,
+					Value: fn(p.Value, p2.Value),
+					Aux:   p.Aux,
+				}
+			},
+		}, nil
+	}
+	return nil, fmt.Errorf("unable to construct transform iterator from %T and %T", lhs, rhs)
+}
+
+func getIteratorDataType(itr Iterator) DataType {
+	switch itr.(type) {
+	case FloatIterator:
+		return Float
+	case StringIterator:
+		return String
+	case BooleanIterator:
+		return Boolean
+	default:
+		return Unknown
+	}
+}
```

Comment:
Can you change the function name from `getIteratorDataType()` to `iteratorDataType()`?

https://golang.org/doc/effective_go.html#Getters